src/basic/path-util.c

```diff
@@ -687,10 +696,18 @@ int find_executable_full(const char *name, bool use_path_envvar, char **ret_file
                 if (!path_is_absolute(element))
                         continue;
 
-                if (!path_extend(&element, name))
+                r = chase_symlinks(element,
+                                   root,
+                                   CHASE_PREFIX_ROOT|CHASE_SAFE|CHASE_NOFOLLOW,
+                                   &path_name,
+                                   /* ret_fd= */ NULL);
+                if (r < 0)
+                        return r;
+
+                if (!path_extend(&path_name, name))
```

Comment:
this needs to happen before chase_symlinks - element + name is what should be passed to chase_symlinks, otherwise this happens:

```
(gdb) n
699	                r = chase_symlinks(element,
(gdb) p element
$14 = 0x55555556a0f0 "/home/luca/npm/bin"
(gdb) p root
$15 = 0x0
(gdb) p name
$16 = 0x555555562d1d "sh"
```

Moreover, errors need to be ignored just like in check_x_access, otherwise not finding a binary under the first PATH is a fatal error, ie something like:

```
                if (!path_extend(&element, name))
                        return -ENOMEM;

                r = chase_symlinks(element,
                                   root,
                                   CHASE_PREFIX_ROOT|CHASE_SAFE|CHASE_NOFOLLOW,
                                   &path_name,
                                   /* ret_fd= */ NULL);
                if (r < 0) {
                        if (r != -EACCES)
                                last_error = r;
                        continue;
                }
```